addons/web/static/tests/legacy/core/owl_dialog_tests.js

```diff
@@ -267,6 +273,58 @@ odoo.define('web.owl_dialog_tests', function (require) {
 
             parent.destroy();
         });
+
+        QUnit.test("Interactions between legacy owl dialogs and new owl dialogs", async function (assert) {
+            assert.expect(7);
+
+            const { legacyEnv, env } = await makeLegacyDialogMappingTestEnv();
+
+            // OwlDialog env
+            patchWithCleanup(Dialog.prototype, {
+                setup() {
+                    this.env = legacyEnv;
+                    this._super();
+                }
+            });
```

Comment:
as far as I understand, you could get rid of this patch by more accurately spawning the legacy dialog (i.e. like we actually do in legacy): `new Dialog().open();`

We can talk about this, as I may be missing something